### PR TITLE
fix(deps): updated use-resize-observer dependency

### LINKED ISSUE #1250: [use-resize-observer] update to v6.1, single ResizeObserver instance
[Internal Watson-IoT/pal-tracking#440](https://github.ibm.com/Watson-IoT/pal-tracking/issues/440)

### Summary

`use-resize-observer` [v6.1 contains a very important performance update](https://github.com/ZeeCoder/use-resize-observer/issues/32) to utilize a single ResizeObserver instance instead of creating a new one with each instance of the hook.

We should update this dep - there is a breaking change regarding the default height/width that we'll need to update `Breadcrumb` to support.

### PR DESCRIPTION
**Pull request by marcJV**
Closes #1250

**Summary**

Updated use-resize-observer dependency

**Change List (commits, features, bugs, etc)**

package.json and yarn.lock

**Acceptance Test (how to verify the PR)**

yarn test

### CODE REVIEW

**Comment by davidicus** [comment]
@marcJV Please fix the issues with travis, update the branch, and, if ready, convert this to a PR.

File: src/components/Breadcrumb/Breadcrumb.test.jsx [hunk 1]
```diff
@@ -50,39 +50,4 @@ describe('Breadcrumb with overflow', () => {
...
-      render(
-        <Breadcrumb {...commonProps} hasOverflow>
-          <BreadcrumbItem href="#">Breadcrumb 1</BreadcrumbItem>
-          <BreadcrumbItem href="#">Breadcrumb 2</BreadcrumbItem>
-          <BreadcrumbItem href="#">Breadcrumb 3</BreadcrumbItem>
-        </Breadcrumb>
-      );
-      expect(console.error).toHaveBeenCalledTimes(1);
-    });
-  });
```
Comment (tay1orjones) [inline src/components/Breadcrumb/Breadcrumb.test.jsx:87]: Why is this being removed? It ensures we only pop the console warning under the proper circumstances. This test was written when we had a bug raised about the warning condition being wrong.

File: src/utils/componentUtilityFunctions.js [hunk 2]
```diff
@@ -312,7 +312,7 @@ export const filterValidAttributes = props =>
 export const browserSupports = api => {
   switch (api) {
     case 'ResizeObserver':
-      return typeof ResizeObserver !== 'undefined';
+      return typeof window.ResizeObserver !== 'undefined';
```
Comment (tay1orjones) [inline src/utils/componentUtilityFunctions.js:315]: Should we nullcheck `window` in the event it's not available in non-browser environments?

```suggestion
      return typeof window?.ResizeObserver !== 'undefined';
```

File: src/internal/UseResizeObserver.jsx [hunk 3]
```diff
@@ -0,0 +1,23 @@
+import useResizeObserver from 'use-resize-observer';
+import warning from 'warning';
+
+import { browserSupports } from '../utils/componentUtilityFunctions';
+
+export const useResize = resizeRef => {
+  if (!browserSupports('ResizeObserver')) {
+    warning(
+      false,
+      'You have set hasOverflow to true, but the current browser does not support ResizeObserver. You will need to include a ResizeObserver polyfill for hasOverflow to function properly.'
```
Comment (bjornalm) [inline src/internal/UseResizeObserver.jsx:10]: I think you need to modify this warning text to fit the generic purpose of this HOC

Comment (marcJV) [reply src/internal/UseResizeObserver.jsx:10]: pushing a fix up right now

File: src/internal/UseResizeObserver.jsx [hunk 4]
```diff
@@ -0,0 +1,23 @@
...
+    );
+    return null;
+  }
+
+  /**
+   * use-resize-observer currently crashes if it is unsupported
+   * we need to conditionally set this hook for now until it is fixed in the library
+   */
+  // eslint-disable-next-line react-hooks/rules-of-hooks
+  useResizeObserver(resizeRef);
```
Comment (bjornalm) [inline src/internal/UseResizeObserver.jsx:20]:
```suggestion
  useResizeObserver({
    useDefaults: false,
    ref: resizeRef,
  });
```

File: src/internal/UseResizeObserver.jsx [hunk 5]
```diff
@@ -0,0 +1,26 @@
...
+
+  /**
+   * use-resize-observer currently crashes if it is unsupported
+   * we need to conditionally set this hook for now until it is fixed in the library
+   */
+  // eslint-disable-next-line react-hooks/rules-of-hooks
+  useResizeObserver({
+    useDefaults: false,
+    ref: resizeRef,
+  });
```
Comment (tay1orjones) [inline src/internal/UseResizeObserver.jsx:23]: Will be fixed via https://github.com/ZeeCoder/use-resize-observer/pull/42

https://github.com/IBM/carbon-addons-iot-react/issues/1426

**Pull request merged**